kitsune/products/migrations/0010_auto_20240624_0616.py

```diff
@@ -0,0 +1,156 @@
+        "customization-firefox-enterprise-environment",
+        "policies-customization-enterprise",
+        "manage-certificates-firefox-enterprise",
+        "policies-overview-enterprise",
+        "manage-settings-policy",
+        "autoconfiguration-enterprise",
+        "hubs-permissions-settings",
+        "thunderbird-controls-and-buttons",
+        "tab-settings",
+        "thunderbird-options-preferences-and-settings",
+        "update-account-settings",
+        "customize-controls-options-and-add-ons",
+    ],
+    "install-and-manage-add-ons": [
+        "install-and-manage-add-ons-firefox",
+        "use-extensions-and-add-ons",
+        "add-ons",
+        "manage-add-ons-enterprise",
+        "customize-thunderbird-add-ons-plugins-and-extensions",
+    ],
+}
+
+
+def move_kb_articles(apps, schema_editor):
+    Topic = apps.get_model("products", "Topic")
+    TopicSlugHistory = apps.get_model("products", "TopicSlugHistory")
+    title = ""
+    for target_slug in TOPICS_TO_MIGRATE.keys():
+
+        match target_slug:
+            case "install-and-update":
+                title = "Install and Update"
+            case "troubleshooting":
+                title = "Troubleshooting"
+            case "protect-your-privacy":
+                title = "Protect your privacy"
+            case "emails":
+                title = "Emails"
+            case "customize-settings-and-preferences":
+                title = "Customize settings and preferences"
+            case "install-and-manage-add-ons":
+                title = "Install and manage add-ons"
+            case _:
+                pass
+
+        source_topics = Topic.objects.filter(slug__in=TOPICS_TO_MIGRATE.get(target_slug, []))
+        for source_topic in source_topics:
+            product = source_topic.product
+            try:
+                target_topic = Topic.objects.get(title=title, slug=target_slug, product=product)
+            except Topic.DoesNotExist:
+                target_topic = Topic.objects.create(
+                    title=title,
+                    slug=target_slug,
+                    product=product,
+                    display_order=source_topic.display_order,
+                )
+
+            # Make sure that target topics are not sub-topics
+            if target_topic.parent:
+                target_topic.parent = None
+                target_topic.save()
+
+            # Migrate KB articles to new topics
+            documents = source_topic.document_set.all()
+            for document in documents:
+                document.topics.add(target_topic)
+                document.topics.remove(source_topic)
+
+            # handle redirects for the old topics
+            if (
+                not TopicSlugHistory.objects.filter(slug=source_topic.slug).exists()
+                and product.visible
+            ):
+                TopicSlugHistory.objects.create(slug=source_topic.slug, topic=target_topic)
+
```

Comment:
Most probably it shouldn't matter but let's err on the side of not visible in case anything slips into the UI. Most probably we shouldn't list anything in the UI if there's a redirect on it.